ludwig/utils/heuristics.py

```diff
@@ -0,0 +1,25 @@
+from ludwig.types import ModelConfigDict
+from ludwig.utils.config_utils import has_pretrained_encoder, has_trainable_encoder, has_unstructured_input_feature
+
+
+def get_auto_learning_rate(config_dict: ModelConfigDict) -> float:
+    """Uses config heuristics to determine an appropriate learning rate.
+
+    The main idea behind the following heuristics is that smaller learning rates are more
+    suitable for features with larger encoders, which are typically used with unstructured features.
+    Note that these are meant to be rough heuristics that are solely based on feature types and the
+    type of the corresponding encoder. More factors could be taken into consideration such as model
+    size, dataset size, batch size, number of features, etc.
+
+    param:
```

Comment:
nit: Call this "Args": https://google.github.io/styleguide/pyguide.html#244-decision